CMakeLists.txt

```diff
@@ -102,6 +106,64 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Generic")
     if(PLATFORM_NAME STREQUAL "nuttx")
         set(PLATFORM_NAME_NUTTX ON)
+        set(TRANSPORT_TYPES "UDP" "TCP" "SERIAL")
+    endif()
+endif()
+
+# Check external transport.
+SET(PLATFORM_EXTERNAL_TRANSPORT_HEADER "" CACHE PATH "External transports header file")
+SET(PLATFORM_EXTERNAL_TRANSPORT "" CACHE PATH "External transports code file")
+
+if( PLATFORM_EXTERNAL_TRANSPORT)
+    set(PLATFORM_EXTERNAL ON)
+
+    set(TRANSPORT_TYPES "SERIAL")
+
+    get_filename_component(PLATFORM_EXTERNAL_TRANSPORT_HEADER_FILENAME ${PLATFORM_EXTERNAL_TRANSPORT_HEADER} NAME)
+    get_filename_component(PLATFORM_EXTERNAL_TRANSPORT_FILENAME ${PLATFORM_EXTERNAL_TRANSPORT} NAME)
+    get_filename_component(PLATFORM_EXTERNAL_TRANSPORT_HEADER_DIRECTORY ${PLATFORM_EXTERNAL_TRANSPORT_HEADER} DIRECTORY)
+endif()
+
+# Transport sources.
+foreach(TRANS_TYPE ${TRANSPORT_TYPES})
+    if(PROFILE_${TRANS_TYPE}_TRANSPORT)
+        string(TOLOWER ${TRANS_TYPE} TT)
+        if(PLATFORM_NAME_LINUX)
+            set(${TRANS_TYPE}_SRCS
+                src/c/profile/transport/${TT}/${TT}_transport.c
+                src/c/profile/transport/${TT}/${TT}_transport_linux.c
+                )
+        elseif(PLATFORM_NAME_WINDOWS)
+            set(${TRANS_TYPE}_SRCS
+                src/c/profile/transport/${TT}/${TT}_transport.c
+                src/c/profile/transport/${TT}/${TT}_transport_windows.c
+                )
+        elseif(PLATFORM_NAME_NUTTX)
+            set(${TRANS_TYPE}_SRCS
+                src/c/profile/transport/${TT}/${TT}_transport.c
+                src/c/profile/transport/${TT}/${TT}_transport_linux.c
+                )
+        elseif(PLATFORM_EXTERNAL)
+            set(${TRANS_TYPE}_SRCS
+                src/c/profile/transport/${TT}/${TT}_transport.c
+                ${PLATFORM_EXTERNAL_TRANSPORT}
+            )
+        endif()
+    endif()
+endforeach()
```

Comment:
This approach changed, now we add the sources when we set the `SRCS`variable. Take a look to line 199.

Reply:
New approach used in external transports.